### PR TITLE
run eslint migration tool (#309)

### PR DESCRIPTION
**Pull request by hypdeb**
Wanted to start contributing but noticed that some of the tooling (TSLint) is deprecated. So I decided to create an issue (#309) to fix it. I ran the automated migration procedure provided by ESLint and this is basically the changes resulting from that.

I'm not a Typescript or VS Code extension developer by trade, I welcome any feedback. In particular I'm a bit worried about the number of changes to the `package-lock.json` file.

### CODE REVIEW

**Review by jfirebaugh — COMMENTED** [review]
Thanks for doing this! I would very happy to merge this if the lint failures are resolved.

I split out the Ubuntu / Node upgrade in https://github.com/bazelbuild/vscode-bazel/pull/312.

**Comment by hypdeb** [comment]
Perfect. I'll fix the issues in the coming weeks and rebase once you've merged #312.

File: src/extension/extension.ts
Comment (jfirebaugh) [inline src/extension/extension.ts:56]: For the no-floating-promises violations, instead of disabling those with `void`, can you use `eslint-disable-next-line`? I think these should be audited at some point as they could constitute bugs, and `eslint-disable-next-line` will be easier to audit than `void`.

Comment (hypdeb) [reply src/extension/extension.ts:56]: Makes sense, I will fix in the coming days.

**Comment by hypdeb** [comment]
@jfirebaugh I reverted my changes to floating promises, disabled the linting instead and created #321 to review those.

**Comment by cameron-martin** [comment]
Can this be merged? I need this to update typescript, so I can use the latest version of `vscode-languageclient` in #317.

**Comment by hypdeb** [comment]
@cameron-martin It's pending another review from a code owner.

**Comment by cameron-martin** [comment]
Yeah I was asking @jfirebaugh.

**Review by jfirebaugh — APPROVED** [review]
Thanks! I fixed up a few things, looks good to go now.

**Pull request merged**
